src/java/cern/accsoft/steering/jmad/tools/response/FullResponseMatrixTool.java

```diff
@@ -1,209 +1,241 @@
+            for (String monitorName : monitorNames) {
+                resultRequest.addElementFilter(monitorName);
+            }
+        } else {
+            /*
+             * if element-names for the response were defined for the range, then we use them
+             */
+            for (String regexp : monitorRegexps) {
+                resultRequest.addElementFilter(regexp);
+            }
+        }
+        resultRequest.addVariable(MadxTwissVariable.NAME);
+        resultRequest.addVariable(MadxTwissVariable.X);
+        resultRequest.addVariable(MadxTwissVariable.Y);
+        resultRequest.addVariable(MadxTwissVariable.KEYWORD);
+
+        TfsResultImpl tfsResult;
+        try {
+            Result result = model.twiss(resultRequest);
+            if (ResultType.TFS_RESULT != result.getResultType()) {
+                throw new JMadModelException("Twiss returned wrong type of result!");
+            }
+            tfsResult = (TfsResultImpl) result;
+        } finally {
+            /* reset strength to old Value */
+            addKickToElement(model, element, plane, -kick);
+        }
+        return tfsResult;
+    }
+
+    private void addKickToElement(JMadModel model, Element element, JMadPlane plane, double kick) throws JMadModelException {
+        if (JMadElementType.CORRECTOR.isTypeOf(element)) {
+            Corrector corrector = (Corrector) element;
+            double oldKick = corrector.getKick(plane);
+            corrector.setKick(plane, oldKick + kick);
+        } else if (JMadElementType.BEND.isTypeOf(element)) {
+            int tiltSign = bendFieldErrorSignFromTilt(element, plane);
+            AddFieldErrors fieldErrorsTask = new AddFieldErrors(element.getName(),
+                    singletonList(kick * tiltSign));
+            model.execute(fieldErrorsTask.compose());
+        } else {
+            throw new JMadModelException("Element '" + element.getName()
+                    + "' is not a corrector or bend! Cannot calc response for this element!");
+        }
+    }
+
+    private int bendFieldErrorSignFromTilt(Element element, JMadPlane plane) throws JMadModelException {
+        double elementTilt = Optional.ofNullable(element.getAttribute("tilt")).orElse(0.0);
+        int tiltSign;
+        if (plane == JMadPlane.H && Math.abs(elementTilt) < BEND_TILT_TOLERANCE) {
+            tiltSign = -1;
+        } else if (plane == JMadPlane.H && Math.abs(Math.abs(elementTilt) - Math.PI) < BEND_TILT_TOLERANCE) {
+            tiltSign = 1;
+        } else if (plane == JMadPlane.V && Math.abs(elementTilt - Math.PI / 2) < BEND_TILT_TOLERANCE) {
+            tiltSign = -1;
+        } else if (plane == JMadPlane.V && Math.abs(elementTilt + Math.PI / 2) < BEND_TILT_TOLERANCE) {
+            tiltSign = 1;
+        } else {
+            String planeAngle = plane == JMadPlane.H ? "0 rad" : "pi/2 rad";
+            throw new JMadModelException("Element '" + element.getName() + "' is a BEND with tilt=" + elementTilt
```

Comment:
Yes, we could offer this, but the point is, each bend can only be used as a corrector in one particular plane (the plane it is in) - so a upright bend can only kick in H and a skew bend can only kick in V. 
By the way, probably I should introduce the same check about HKICKER (H only) and VKICKER (V only) ...

Reply:
Ok, I see for H/VKICKER there is already a check in `Corrector`, but it 'only' logs a warning and does not throw ...

Reply:
I think in madx, an HKicker can even produce a vertical kick ;-) or?

No particular opinion here ...

Reply:
At least in JMad (using the `setKick(Plane, double)` method) it can't - for the wrong plane JMad will issue a WARN level message and ignore the kick ...

Anyway, I will leave it for the moment and keep it in mind if I need it when making the changes in Aloha ...